tests/unit_tests/backends/test_collisions_methods.py

```diff
@@ -207,3 +207,33 @@ def test_scale_prob_for_adaptive_sdm_gamma(
                 )
         np.testing.assert_array_almost_equal(_gamma.to_ndarray(), expected_gamma)
         np.testing.assert_array_equal(_n_substep, np.asarray(expected_n_substep))
+
+    @staticmethod
+    # pylint: disable=redefined-outer-name
+    def test_cell_caretaker(backend_class):
+        # Arrange
+        backend = backend_class()
+        idx = [0, 3, 2, 4]
+
+        cell_start = backend.Storage.from_ndarray(np.asarray([-1, -1]))
+        _idx = make_Index(backend).from_ndarray(np.asarray(idx, dtype=np.int64))
+
+        multiplicity = make_IndexedStorage(backend).from_ndarray(
+            _idx, np.asarray([1, 1, 1, 1])
+        )
+        _idx.remove_zero_n_or_flagged(multiplicity)
+
+        cell_id = make_IndexedStorage(backend).from_ndarray(
+            _idx, np.asarray([0, 0, 0, 0])
+        )
+        cell_idx = make_Index(backend).from_ndarray(np.asarray([0]))
+
+        sut = backend.make_cell_caretaker(
+            _idx.shape, _idx.dtype, len(cell_start), scheme="counting_sort"
```

Comment:
here, I assume, we would benefit (also from code coverage perspective) from testing as well the `scheme=default` and the `scheme=counting_sort_parallel` variants (even if, for now, these would be marked as XFAIL for GPU)

Reply:
and if the GPU backend ignores this setting, let's raise an exception if the user specifies something else than `default`?